lib.js

```diff
@@ -55,34 +58,62 @@ module.exports = ({
       identifier, it is recommended to customize the package identifier.`);
   }
 
-  return Promise.all(templates.filter((template) => {
-    if (template.platform) {
-      return (platforms.indexOf(template.platform) >= 0);
-    }
-
-    return true;
-  }).map((template) => {
-    if (!template.name) {
-      return Promise.resolve();
-    }
-
-    const args = {
-      name: `${prefix}${pascalCase(name)}`,
-      moduleName: `${modulePrefix}-${paramCase(name)}`,
-      packageIdentifier,
-      namespace: namespace || pascalCase(name).split(/(?=[A-Z])/).join('.'),
-      platforms,
-      githubAccount,
-      authorName,
-      authorEmail,
-      license,
-    };
+  return Promise
```

Comment:
I think we don't need the Promise.resolve here, just return the createFolder

Furthermore I think in all of the early returns for the `!generateExample` checks we should return a Promise.reject if we consume the promise that is returned from this funciton or a Promise.resolve in case we see it as success.

Reply:
Yea, that's a good improvement. I move the `createFolder` out.